README.md

```diff
@@ -1,50 +1,86 @@
-Yojson: low-level JSON library for OCaml
-========================================
+Yojson: JSON library for OCaml
+==============================
 
 [![Build Status](https://travis-ci.org/ocaml-community/yojson.svg?branch=master)](https://travis-ci.org/ocaml-community/yojson)
 
-_This library is for manipulating the json AST directly. For mapping between OCaml types and json, we recommend [atdgen](https://github.com/ahrefs/atd)._
+This library parses complete JSON files into a nested OCaml data structure,
+similarly to DOM parsers for XML. For stream processing in a SAX-like fashion
+check out [jsonm](https://erratique.ch/software/jsonm).
+
 
 Library documentation
 ---------------------
 
 Currently at https://ocaml-community.github.io/yojson/
 
-Design goals
-------------
 
-* reducing inter-package dependencies by the use of polymorphic
-  variants for the JSON tree type
+Examples
+--------
+
+A simple example on how to parse JSON from a string literal.
+
+```ocaml
+let json_string = {|
+  {"number" : 42,
+   "string" : "yes",
+   "list": ["for", "sure", 42]}|}
+(* val json_string : string *)
+
+let json = Yojson.Safe.from_string json_string
```

Comment:
I wonder if we shouldn't advocate the use of `Basic` rather than safe as it represents actual JSON. That will make it easier to drop the other types if everybody starts using the "right" one.

Reply:
I think the least invasive way to go forward is to remove tuples and variants from `Safe` since that's the variant that nearly all users (including all PPX rewriters) have agreed upon.

But for now I would suggest `Safe` as the "main" mode that we advertise, just to reduce confusion and avoid people running into a problem where they have `Yojson.Basic.t` in their code but get `Yojson.Safe.t` from other libraries.

Reply:
Yeah that's a good point, especially for derivers! Let's keep it that way!